Extend packet information in Lua alert module

miska opened the pull request:
I'm playing with snort and I want to customize processing of alerts. The simplest way to do so seems to be writing Lua alert module. Unfortunately there seems to be some pieces of information missing and given there is TODO mentioned in source code to add those, I went ahead and added some. Tested locally and it seems to work.

BKPepe commented:
@mjs717 or @Xiche, would it be possible to take a look at this, please?

Xiche commented:
> @mjs717 or @Xiche, would it be possible to take a look at this, please?

Sorry, I'm not really involved with this project the way I used to be.  Conceptually, it certainly looks like a step in the right direction for making the LuaJIT logger a bit more realized and useful.  From an actual implementation perspective, it has a number of issues like using non-thread-local static buffers for address string storage (not safe with multiple packet threads, not to mention the stringification should probably be more of a JIT operation) and magic numbers all over the place for buffer sizes.  Also, at some point in the process, it's probably better to expand the APIs available to the Lua interface rather than just shoving more data in one of the existing retrievable structs (more like AppID).

You probably want @snortadmin to look at this more than anyone else.

miska commented:
Rebased and fixed the mentioned issues. The only thing that was mentioned and is still there is stringification in C. From users point of view, it makes sense to me to provide an easy way how to access those data and TODO I removed talked about adding such information into the structure. I understand that only when needed approach would be probably more effective as you could avoid stringification, but I guess that would require changing the Lua API which would be much bigger change and I have no idea what are your future plans for this.
